include/boost/math/special_functions/detail/bernoulli_details.hpp

```diff
@@ -201,18 +201,9 @@ struct fixed_vector : private std::allocator<T>
    }
    ~fixed_vector()
    {
-#ifdef BOOST_NO_CXX11_ALLOCATOR
       for(unsigned i = 0; i < m_used; ++i)
          this->destroy(&m_data[i]);
       this->deallocate(m_data, m_capacity);
-#else
```

Comment:
I think you've deleted the wrong branch here?

This will break C++17 as well I believe.